ext/mbstring/mbstring.c

```diff
@@ -2358,8 +2358,11 @@ MBSTRING_API char *php_mb_convert_encoding_ex(const char *input, size_t length,
 
 	/* initialize converter */
 	convd = mbfl_buffer_converter_new(from_encoding, to_encoding, string.len);
-	/* If this assertion fails this means some memory allocation failure which is a bug */
-	ZEND_ASSERT(convd != NULL);
+
+	if (convd == NULL) {
+		php_error_docref(NULL, E_WARNING, "Cannot convert between the specified text encodings");
+		return NULL;
+	}
```

Comment:
I don't think this error condition is tested, which encoding pairs does it occur for?

Reply:
Thanks for pointing that out. I will have to add a test.

When I was trying things out, I found a case where it was possible to crash PHP. Now I'm trying to remember what pair it was. It may have involved `8bit` and `7bit`.

Let me see if I can dig up that information and add a test.

Reply:
Or it might have involved using `wchar` as an explicit encoding... but you've (thankfully) removed that one now...

Reply:
Maybe let me just remove that commit for now. If I later remember what the pair in question was, I can do this again (but with proper testing).

It may have very well involved passing `"wchar"` as the encoding.